pallets/crowdloan/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+//! # Crowdloan rewards Pallet
+//!
+//! Provides means of registering the contributors to the KILT crowdloan.
+//!
+//! - [`Config`]
+//! - [`Call`]
+//! - [`Pallet`]
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//! - `set_admin_account` - Set the account that is allow to register and delete
+//!   contribution entries into/from this pallet's storage.
+//! - `set_new_contribution` - Add or replace a crowdload contribution, which
+//!   contains the contributor's address and the contributed amount.
+//! - `remove_contribution` - Remove a contribution entry from the pallet
+//!   storage.
+//!
+//! ## Genesis config
+//!
+//! The crowdloan contributions pallet depends on the [`GenesisConfig`].
+//!
+//! ## Assumptions
+//!
+//! - At any time, there is one and only one admin account which can manage the
+//!   pallet storage.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+pub(crate) mod mock;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub use crate::pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{Currency, StorageVersion},
+	};
+	use frame_system::{pallet_prelude::*, EnsureOneOf, EnsureRoot, EnsureSigned, WeightInfo};
+	use sp_runtime::{
+		traits::{BadOrigin, StaticLookup},
+		Either,
+	};
+
+	pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+	pub type BalanceOf<T> = <<T as Config>::Currency as Currency<AccountIdOf<T>>>::Balance;
+
+	pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Currency type.
+		type Currency: Currency<AccountIdOf<Self>>;
+		/// Overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub admin_account: AccountIdOf<T>,
```

Comment:
C: I think admin is a strong word. A block chain should not have an admin. Maybe rename to `registrar`?

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/264/commits/7e18b4424602754004a8c33b0f8508105d1112a9